File: apps/java/src/main/java/com/workshop/runner/RepoRoot.java
package com.workshop.runner;

import java.io.File;

/** Locates the repository root holding the shared prompts/, schemas/, datasets/. */
public final class RepoRoot {
    private RepoRoot() {}

    public static File find() {
        String env = System.getenv("WORKSHOP_REPO_ROOT");
        if (env != null && !env.isBlank()) return new File(env);

        File dir = new File(System.getProperty("user.dir"));
        while (dir != null) {
            if (new File(dir, "prompts").isDirectory() && new File(dir, "schemas").isDirectory()) {
                return dir;
            }
            dir = dir.getParentFile();
        }
        return new File(System.getProperty("user.dir"));
    }
}

File: apps/csharp/src/WorkshopRunner/RepoRoot.cs
namespace WorkshopRunner;

/// <summary>Locates the repository root that holds the shared prompts/, schemas/, datasets/.</summary>
public static class RepoRoot
{
    public static string Find()
    {
        var env = Environment.GetEnvironmentVariable("WORKSHOP_REPO_ROOT");
        if (!string.IsNullOrWhiteSpace(env)) return env;

        var dir = new DirectoryInfo(AppContext.BaseDirectory);
        while (dir is not null)
        {
            if (Directory.Exists(Path.Combine(dir.FullName, "prompts")) &&
                Directory.Exists(Path.Combine(dir.FullName, "schemas")))
                return dir.FullName;
            dir = dir.Parent;
        }
        // Fallback: current working directory.
        return Directory.GetCurrentDirectory();
    }
}
